libbeat/statestore/backend/backend.go

```diff
@@ -68,4 +68,8 @@ type Store interface {
 	// is assumed to be invalidated once fn returns
 	// The loop shall return if fn returns an error or false.
 	Each(fn func(string, ValueDecoder) (bool, error)) error
+
+	// Sets the store ID when the full input configuration is acquired
+	// This is needed in order to support Elasticsearch state store naming convention based on the input ID
+	SetID(id string)
```

Comment:
The comment implies this means there will be a store per input, would this make more sense being a required argument when the store is constructed? You can't have an interface that includes a constructor, but that is where it feels like it belongs.

Reply:
Trying to workaround the current code interfaces that are already in place here. That's why had to add additional function to be able to set the ID when the input configuration is actually received and the ID is available.